public/app/plugins/panel/timeseries/migrations.ts

```diff
@@ -46,10 +52,28 @@ export const graphPanelChangedHandler: PanelTypeChangedHandler = (
 ) => {
   // Changing from angular/flot panel to react/uPlot
   if (prevPluginId === 'graph' && prevOptions.angular) {
-    const { fieldConfig, options } = graphToTimeseriesOptions({
+    const { fieldConfig, options, annotations } = graphToTimeseriesOptions({
       ...prevOptions.angular,
       fieldConfig: prevFieldConfig,
+      panel: panel,
     });
+
+    const dashboard = getDashboardSrv().getCurrent();
+    if (dashboard && annotations.length > 0) {
+      dashboard.annotations.list = [...dashboard.annotations.list, ...annotations];
+      console.log('updating annotations from migration', panel, dashboard.annotations);
+
+      // Trigger a full dashbaord refresh ???
+      if (dashboardRefresher == null) {
+        console.log('scheduling a refresh');
+        dashboardRefresher = setTimeout(() => {
+          console.log('trigger refresh for annotations');
+          dashboard.startRefresh({ refreshAll: true, panelIds: [] });
```

Comment:
@dprokop @torkelo -- any suggestions on how to trigger the annotations to rerun.  This was my hack, but not working consistently

Reply:
@ryantxu the problem here is that this will be triggered multiple times if you have many panels with time regions on dashboard. So maybe need some kind of debounce function for this. But ideally it should only retrigger annotations. Think it could be better if this was something like (But still need to figure out the debounce), the other problem I guess is that when loading a dashboard this is kind of too late. The annotation queries might already be running. Is there a way to do these migrations in the DahboardMigrator instead so they happen before dashboard is initialized and annotation queries are issued? 

```ts
getDashboardQueryRunner().run({dashboard, range: getTimeSrv().getTimeRange()}
```